client/app/pages/dashboards/hooks/useDashboard.js

```diff
@@ -130,6 +130,14 @@ function useDashboard(dashboardData) {
 
   const loadDashboard = useCallback(
     (forceRefresh = false, updatedParameters = []) => {
+      // filter widgets to show from all the widgets according to the current parameters
+      dashboardRef.current.widgets = dashboardRef.current.saved_all_widgets;
+      dashboardRef.current.widgets = getAllowedWidgetsForCurrentParam(
+        dashboardRef.current.getParametersDefs(),
+        dashboardRef.current.allowed_widgets,
+        dashboardRef.current.widgets
+      );
```

Comment:
Do we need to first set the `dashboardRef.current.widgets = dashboardRef.current.saved_all_widgets;` and then do it again? Can't we do:

```suggestion
      dashboardRef.current.widgets = getAllowedWidgetsForCurrentParam(
        dashboardRef.current.getParametersDefs(),
        dashboardRef.current.allowed_widgets,
        dashboardRef.current.saved_all_widgets
      );
```

Reply:
thanks, i updated it in [the latest commit ](https://github.com/metr-systems/redash/pull/3/commits/625fe23e8df46b392c8e47e63e335e85996180ca)